### PR TITLE
Ruby on Rails capstone project - Budget app

### PR DESCRIPTION
**Pull request by ShoiraTa**
### Ruby on Rails capstone project - Budget app

Link to deployed page: https://budget-app-003.herokuapp.com/users/2/groups
Video presentation : https://www.loom.com/share/cf99fe8d41f34549afe2cb227d8cec03 

In this PR I created a web app that allows the user to:

- register and log in, so that the data is private to them.
- introduce new transactions associated with a category.
- see the money spent on each category.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,64 @@
...
+
+Feel free to check the [issues page](../../issues/).
+
+## Show your support
+
+Give a ⭐️ if you like this project!
+
+## Acknowledgments
+
+- Microverse
```
Comment (DeliceLydia) [inline README.md:60]: - [ ] Please give a credit to [Gregoire Vella on Behance](https://www.behance.net/gregoirevella) the [author of the original design](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=),

File: spec/models/expenditure_spec.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    context 'when valid' do
+      it { expect(expenditure).to be_valid }
+    end
+
+    it 'Without name it is invalid ' do
+      expenditure.name = nil
+      expect(expenditure).to_not be_valid
+    end
+  end
+end
```
Comment (DeliceLydia) [inline spec/models/expenditure_spec.rb:18]: - [ ] Test are failing please check an error you are having in this file
tets

File: app/models/expenditure.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  validates :amount, presence: true
+
+  def total_expenditures(group_id)
+    total = 0
+    Expenditure.where(group_id:).each do |expenditure|
+      total += expenditure.amount
+    end
+    total
+  end
+end
```
Comment (DeliceLydia) [inline app/models/expenditure.rb]: - [ ] Apparently, there is a small issue in this `expenditure.rb` file which is causing the server to not run please address that
err

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hello @ShoiraTa, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there! Highlights are:
- App is running well on Heroku
- Awesome video demo
- Good README
- Followed git workflow


## Required Changes ♻️
_Check the comments under the review._



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)

File: app/models/expenditure.rb (hunk 3 above)
Comment (ahmetbozaci) [reply app/models/expenditure.rb]: - [ ] When ı try to add a category ı am getting an error. Please fix it. It is just a syntax error. You can fix it with writing like that.
`    Expenditure.where(group_id: group_id).each do |expenditure|
`
I see it is working in the Heroku link but maybe you deleted this after pushing to Heroku.
image

File: spec/models/expenditure_spec.rb (hunk 2 above)
Comment (ahmetbozaci) [reply spec/models/expenditure_spec.rb:18]: - [ ] If you fix the syntax error in the expenditure model, model tests will pass. Please fix it.

File: spec/views/expenditure_spec.rb [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+      click_link 'group'
+      expect(page).to have_content('new Expenditure')
+      expect(page).to have_content('Total spent: $40')
+    end
+
+    scenario 'I can open a a page to add a new expenditure' do
+      click_link 'group'
+      click_link 'Add a new expenditure'
+      expect(page).to have_content('NEW EXPENDITURE')
+    end
```
Comment (ahmetbozaci) [inline spec/views/expenditure_spec.rb:33]: - [ ] If you fix the error in the model, rspec error will fix also. 

image

**Review by ahmetbozaci — CHANGES_REQUESTED** [review]
# STATUS: Changes Required  ♻️


@ShoiraTa  ✋🏼 
#### Great work on making the changes requested by a previous reviewer 👏 
#### You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.
___
### ♻️ Request changes ♻️

- [ ] _Check the comments under the review._

___

### Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
___

### Cheers and Happy coding! 💻 👏
almost


___

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form.](https://airtable.com/shrQAqnBwek5a0O0s)_


_If anything seems to be unclear to you, reach out and ask anything by commenting on this Pull Request and mentioning me as [ahmetbozaci](https://github.com/ahmetbozaci), I will receive a notification and will respond._
___
[GitHub](https://github.com/ahmetbozaci) | [LinkedIn](https://linkedin.com/in/ahmetbozaci) | [Twitter](https://twitter.com/ahmtbozaci)

_Reviewed by [Ahmet Bozacı ](https://github.com/ahmetbozaci)_

File: README.md (hunk 1 above)
Comment (ahmetbozaci) [reply README.md:60]: @ShoiraTa 

- [ ] You should add this license in the license section

Comment (ShoiraTa) [reply README.md:60]: @ahmetbozaci thank you for review, but this is not a license. It is a credit to the author of the design, should I still add in licence section?

File: spec/models/expenditure_spec.rb (hunk 2 above)
Comment (ShoiraTa) [reply spec/models/expenditure_spec.rb:18]: when I run rubocop -A, it deleted part of the code, issue is addressed

File: app/models/expenditure.rb (hunk 3 above)
Comment (ShoiraTa) [reply app/models/expenditure.rb]: I had `Expenditure.where(group_id: group_id)` but rubocop decided to get of one `group_id` 😆

File: README.md (hunk 1 above)
Comment (ahmetbozaci) [reply README.md:60]: @ShoiraTa Yeah you are right. You can keep it in Acknowledgements.

File: app/models/expenditure.rb (hunk 3 above)
Comment (ahmetbozaci) [reply app/models/expenditure.rb]: @ShoiraTa Instead of group_id you can just write id. It won't be a problem.

**Comment by ShoiraTa** [comment]
I added credits on readme's Acknowledgments section since it is not a license, we were required to give credits:
Screenshot 2022-03-24 at 00 10 11
I have MIT license link up and running in license section. 

All specs are running successfully

File: README.md (hunk 1 above)
Comment (ShoiraTa) [reply README.md:60]: @ahmetbozaci thank you

File: app/models/expenditure.rb (hunk 3 above)
Comment (ShoiraTa) [reply app/models/expenditure.rb]: yeah I did that and pushed 👍🏻

**Review by Dipeshtwis — APPROVED** [review]
Hi @ShoiraTa, 

In my opinion, your project is ready for presentation!  There is nothing else to say other than... it's time to merge it :shipit: 


Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**